crates/status/src/status_manager.rs

```diff
@@ -16,93 +20,147 @@ pub enum StatusError {
+    ///
+    /// A `StatusChannel` containing a sender and receiver for the provided states.
+    pub fn new(cl_state: ClientState, l1_status: L1Status, ch_state: Option<Chainstate>) -> Self {
+        let (cl_tx, cl_rx) = watch::channel(cl_state);
+        let (l1_tx, l1_rx) = watch::channel(l1_status);
+        let (chs_tx, chs_rx) = watch::channel(ch_state);
+
+        let sender = Arc::new(StatusSender {
+            cl: cl_tx,
+            l1: l1_tx,
+            chs: chs_tx,
+        });
+        let receiver = Arc::new(StatusReceiver {
+            cl: cl_rx,
+            l1: l1_rx,
+            chs: chs_rx,
+        });
+
+        Self { sender, receiver }
+    }
+
+    // Receiver methods
+
+    /// Gets the latest [`LocalL1State`].
+    pub fn l1_view(&self) -> LocalL1State {
+        self.receiver.cl.borrow().l1_view().clone()
     }
 
-    pub fn csm(&self) -> &watch::Receiver<CsmStatus> {
-        &self.csm
+    /// Gets the latest [`SyncState`].
+    pub fn sync_state(&self) -> Option<SyncState> {
+        self.receiver.cl.borrow().sync().cloned()
     }
 
-    pub fn cl(&self) -> &watch::Receiver<ClientState> {
-        &self.cl
+    /// Gets the latest operator table.
+    pub fn operator_table(&self) -> Option<OperatorTable> {
+        self.receiver
+            .chs
+            .borrow()
+            .clone()
+            .map(|chs| chs.operator_table().clone())
     }
 
-    pub fn l1(&self) -> &watch::Receiver<L1Status> {
-        &self.l1
+    /// Gets the latest deposits table.
+    pub fn deposits_table(&self) -> Option<DepositsTable> {
+        self.receiver
+            .chs
+            .borrow()
+            .clone()
+            .map(|chs| chs.deposits_table().clone())
     }
 
-    pub fn chs(&self) -> &watch::Receiver<Option<Chainstate>> {
-        &self.chs
+    /// Gets the latest [`L1Status`].
+    pub fn l1_status(&self) -> L1Status {
+        self.receiver.l1.borrow().clone()
+    }
+
+    /// Waits until there's a new client state and returns the client state.
+    pub async fn wait_for_client_change(&self) -> Result<ClientState, RecvError> {
+        let mut s = self.receiver.cl.clone();
+        s.changed().await?;
+        let state = s.borrow().clone();
+        Ok(state)
     }
-}
 
-impl StatusTx {
-    pub fn update_chain_state(&self, post_state: &Chainstate) {
-        if self.chs.send(Some(post_state.clone())).is_err() {
+    /// Waits until genesis and returns the client state.
+    pub async fn wait_until_genesis(&self) -> Result<ClientState, RecvError> {
+        let mut rx = self.receiver.cl.clone();
+        loop {
+            if rx.borrow().has_genesis_occurred() {
+                return Ok(rx.borrow().clone());
+            }
+            rx.changed().await?;
+        }
+    }
+
+    // Sender methods
+
+    /// Sends the updated `Chainstate` to the chain state receiver. Logs a warning if the receiver
+    /// is dropped.
+    pub fn update_chainstate(&self, post_state: &Chainstate) {
```

Comment:
This take ownership of the chainstate so it should take it by value instead of by ref.

Reply:
Done.